### PR TITLE
Add error page for paths which are not mapped

### PR DESCRIPTION
**Pull request by jphillips2121**
If I path is not mapped for the `report-a-discrepancy` root then an error page will be displayed

Resolves: FAML-390

### CODE REVIEW

File: server/views/report/error.njk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+                        <p>
+                            If this problem continues please
+                            <a href="mailto: enquiries@companieshouse.gov.uk">email us</a>
+                        </p>
+                        </div>
+                    </div>
+                </div>
+            </main>
+        </div>
+{% endblock %}
```
Comment (mwejuli) [inline server/views/report/error.njk:22]: Shouldn't our HTML content be similar to [this page](https://design-system.service.gov.uk/patterns/page-not-found-pages/)?

File: server/routes/report.js [hunk 2]
```diff
@@ -237,4 +237,9 @@ router.get('/report-a-discrepancy/confirmation', (req, res) => {
   res.render(`${routeViews}/confirmation.njk`, { title: 'PSC discrepancy submitted' });
 });
 
+router.get('/report-a-discrepancy/**', (req, res) => {
```
Comment (mwejuli) [inline server/routes/report.js:240]: perhaps this logic sits best in `server/router.js`

**Review by mwejuli — CHANGES_REQUESTED** [review]

File: server/views/report/error.njk (hunk 1 above)
Comment (jphillips2121) [reply server/views/report/error.njk:22]: We discussed it the other day and decided to keep any error screens to the generic technical difficulties screen that we use on other services, like the one used in Java [here](https://github.com/companieshouse/common-web-java/blob/master/src/main/resources/templates/error.html)

Comment (mwejuli) [reply server/views/report/error.njk:22]: ok, my understanding is "generic technical difficulties" is quite different from "missing page" ... but happy to approve if @sroberts2 thinks we should leave as is... 😄

**Review by mwejuli — APPROVED** [review]

**Pull request merged**